qualtran/bloqs/factoring/ecc/ec_phase_estimate_r.py

```diff
@@ -64,11 +66,21 @@ def build_composite_bloq(self, bb: 'BloqBuilder', x: Soquet, y: Soquet) -> Dict[
         for i in range(self.n):
             ctrl[i], x, y = bb.add(ECAddR(n=self.n, R=2**i * self.point), ctrl=ctrl[i], x=x, y=y)
 
-        bb.add(MeasureQFT(n=self.n), x=ctrl)
+        ctrls = bb.join(np.array(ctrl), dtype=QUInt(bitsize=self.n))
+        ctrls = bb.add(QFTTextBook(bitsize=self.n), q=ctrls)
+        ctrl = bb.split(ctrls)
```

Comment:
I think we should keep `MeasureQFT` factored out. 1) makes the circuit cleaner at high level of abstraction 2) we'd want to actually do the semi-classical qft which sortof requires that you're measuring it.

You can just implement the decomposition for MeasureQFT as the code you've inlined here